mlflow/models/evaluation/utils/trace.py

```diff
@@ -0,0 +1,92 @@
+import contextlib
+import inspect
+import logging
+
+import mlflow
+from mlflow.ml_package_versions import FLAVOR_TO_MODULE_NAME
+from mlflow.utils.autologging_utils import (
+    AUTOLOGGING_CONF_KEY_IS_GLOBALLY_CONFIGURED,
+    AUTOLOGGING_INTEGRATIONS,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def configure_autologging_for_evaluation(enable_tracing: bool = True):
+    flavor_to_original_config = {}
+    trace_enabled_flavors = []
+    for flavor in FLAVOR_TO_MODULE_NAME:
+        try:
+            if autolog := _get_autolog_function(flavor):
+                original_config = AUTOLOGGING_INTEGRATIONS.get(flavor, {}).copy()
+
+                # "globally_configured" is a special key that is added to the config when it was
+                # enabled via mlflow.autolog() rather than each flavor's autolog() function.
+                # It is not present in the autolog()'s signature, so we need to remove it.
+                original_config.pop(AUTOLOGGING_CONF_KEY_IS_GLOBALLY_CONFIGURED, None)
+
+                # If autologging is explicitly disabled, do nothing.
+                if original_config.get("disable", False):
+                    continue
+
+                elif enable_tracing and _is_trace_autologging_supported(flavor):
+                    # set all log_xyz params to False except log_traces
+                    new_config = {
+                        k: False if k.startswith("log_") else v for k, v in original_config.items()
+                    }
+                    new_config = {**new_config, "log_traces": True, "silent": True}
```

Comment:
```suggestion
                    new_config |= {"log_traces": True, "silent": True}
```

new syntax in python 3.9

Reply:
or:
```suggestion
                    new_config.update({"log_traces": True, "silent": True})
```

Reply:
Ah yes, `|=` is much cleaner!